packages/react/src/useTable.tsx

```diff
@@ -45,20 +45,35 @@ export interface TableOptions {
   fields?: string[];
 }
 
+export type TableData<Data> =
+  | {
+      columns: TableColumn[];
+      rows: Record<string, ColumnValueType>[];
+      data: Data;
+      metadata: ModelMetadata;
+    }
+  | {
+      columns: null;
+      rows: null;
+      data: null;
+      metadata: null;
+    };
+
 export type TableResult<Data> = [
-  (
-    | {
-        columns: TableColumn[];
-        rows: Record<string, ColumnValueType>[];
-        data: Data;
-        page: TablePagination;
-        metadata: ModelMetadata;
-        fetching: boolean;
-        error?: ErrorWrapper;
-      }
-    | { columns: null; rows: null; data: null; metadata: null; page: TablePagination; fetching: boolean; error?: ErrorWrapper }
-  ),
-  (opts?: Partial<OperationContext>) => void
+  {
+    table: TableData<Data> & {
+      page: TablePagination;
```

Comment:
I don't think I'm 100% on board for grouping these under a table key -- I think it just makes it a bit more annoying to destructure them when you're consuming the hook.

I do think all the search props can be grouped together tho.

Reply:
Updated such that the search stuff is grouped together and the previous variables are untouched